### PR TITLE
[zh-cn] Update 'Events'

### PR DESCRIPTION
**Pull request by jasonren0403**

### CODE REVIEW

File: files/zh-cn/learn/javascript/building_blocks/events/index.md [hunk 1]
```diff
@@ -29,331 +28,329 @@ slug: Learn/JavaScript/Building_blocks/Events
...
   e.target.style.backgroundColor = rndCol;
   console.log(e);
 }
 
-btn.addEventListener('click', bgChange);
+btn.addEventListener("click", bgChange);
 ```
 
-> **备注：** 您可以在 Github 上查看这个示例的 [完整代码](https://github.com/mdn/learning-area/blob/master/javascript/building-blocks/events/random-color-eventobject.html) ，或者在这里查看 [实时演示](http://mdn.github.io/learning-area/javascript/building-blocks/events/random-color-eventobject.html)。
+> **备注：** 你可以在 GitHub 上查看这个示例的[完整代码](https://github.com/mdn/learning-area/blob/main/javascript/building-blocks/events/random-color-eventobject.html) ，或者在这里查看[实时演示](http://mdn.github.io/learning-area/javascript/building-blocks/events/random-color-eventobject.html)。
```
Comment (jasonren0403) [inline files/zh-cn/learn/javascript/building_blocks/events/index.md:297]:
```suggestion
> **备注：** 你可以在 GitHub 上查看这个示例的[完整代码](https://github.com/mdn/learning-area/blob/main/javascript/building-blocks/events/random-color-eventobject.html) ，或者在这里查看[实时演示](https://mdn.github.io/learning-area/javascript/building-blocks/events/random-color-eventobject.html)。
```

File: files/zh-cn/learn/javascript/building_blocks/events/index.md [hunk 2]
```diff
@@ -365,157 +362,389 @@ div {
...
+    para.textContent = "You need to fill in both names!";
   }
+});
+```
+
+显然，这是一种非常弱的表单验证——例如，用户输入空格或数字提交表单，表单验证并不会阻止用户提交——但对于演示来说已经足够。输出如下：
+
+{{ EmbedLiveSample('阻止默认行为', '100%', 180, "", "") }}
+
+> **备注：** 你可以在 GitHub 上查看这个示例的完整源代码 [preventdefault-validation.html](https://github.com/mdn/learning-area/blob/main/javascript/building-blocks/events/preventdefault-validation.html)，也可以[在线运行它](http://mdn.github.io/learning-area/javascript/building-blocks/events/preventdefault-validation.html)。
```
Comment (jasonren0403) [inline files/zh-cn/learn/javascript/building_blocks/events/index.md:385]:
```suggestion
> **备注：** 你可以在 GitHub 上查看这个示例的完整源代码 [preventdefault-validation.html](https://github.com/mdn/learning-area/blob/main/javascript/building-blocks/events/preventdefault-validation.html)，也可以[在线运行它](https://mdn.github.io/learning-area/javascript/building-blocks/events/preventdefault-validation.html)。
```

File: files/zh-cn/learn/javascript/building_blocks/events/index.md [hunk 3]
```diff
@@ -29,331 +28,329 @@ slug: Learn/JavaScript/Building_blocks/Events
...
 
-这个 [`onclick`](/zh-CN/docs/Web/API/GlobalEventHandlers/onclick) 是被用在这个情景下的事件处理器的属性，它就像 button 其他的属性（如 [`btn.textContent`](/zh-CN/docs/Web/API/Node/textContent), or [`btn.style`](/zh-CN/docs/Web/API/HTMLElement/style)), 但是有一个特别的地方——当您将一些代码赋值给它的时候，只要事件触发代码就会运行。
+### 监听其他事件
 
-您也可以将一个有名字的函数赋值给事件处理参数（正如我们在 [Build your own function](/zh-CN/docs/Learn/JavaScript/Building_blocks/Build_your_own_function) 中看到的），下面的代码也是这样工作的：
+有许多不同的事件可以由一个按钮元素来触发。让我们来做个实验。
 
-```js
-const btn = document.querySelector('button');
+首先，在本地创建 [random-color-addeventlistener.html](https://github.com/mdn/learning-area/blob/main/javascript/building-blocks/events/random-color-addeventlistener.html) 的复制，并在浏览器中打开。这只是我们已经玩过的简单的随机颜色示例的一个副本。现在试着依次将 `click` 改为以下不同的值，并观察示例中的结果：
```
Comment (yin1999) [inline files/zh-cn/learn/javascript/building_blocks/events/index.md:127]:
```suggestion
首先，在本地创建 [random-color-addeventlistener.html](https://github.com/mdn/learning-area/blob/main/javascript/building-blocks/events/random-color-addeventlistener.html) 的副本，并在浏览器中打开。这只是我们已经玩过的简单的随机颜色示例的一个副本。现在试着依次将 `click` 改为以下不同的值，并观察示例中的结果：
```

File: files/zh-cn/learn/javascript/building_blocks/events/index.md [hunk 4]
```diff
@@ -29,331 +28,329 @@ slug: Learn/JavaScript/Building_blocks/Events
...
 }
 
-btn.addEventListener('click', bgChange);
+btn.onclick = bgChange;
 ```
 
-> **备注：** 您可以在[Github](https://github.com/mdn/learning-area/blob/master/javascript/building-blocks/events/random-color-addeventlistener.html)上找到这个示例的完整源代码 (也可以 [在线运行](http://mdn.github.io/learning-area/javascript/building-blocks/events/random-color-addeventlistener.html))。
-
-在 `addEventListener()` 函数中，我们具体化了两个参数——我们想要将处理器应用上去的事件名称，和包含我们用来回应事件的函数的代码。注意将这些代码全部放到一个匿名函数中是可行的：
+对于事件处理器属性，你不能为一个事件添加一个以上的处理程序。例如，你可以在一个元素上多次调用 `addEventListener('click', handler)`，并在第二个参数中指定不同的功能：
```
Comment (yin1999) [inline files/zh-cn/learn/javascript/building_blocks/events/index.md:222]:
```suggestion
对于事件处理器属性，你不能为一个事件添加一个以上的处理程序。例如，你可以在一个元素上多次调用 `addEventListener('click', handler)`，并在第二个参数中指定不同的函数：
```

File: files/zh-cn/learn/javascript/building_blocks/events/index.md [hunk 5]
```diff
@@ -365,157 +362,389 @@ div {
...
     e.preventDefault();
-    para.textContent = 'You need to fill in both names!';
+    para.textContent = "You need to fill in both names!";
   }
+});
+```
+
+显然，这是一种非常弱的表单验证——例如，用户输入空格或数字提交表单，表单验证并不会阻止用户提交——但对于演示来说已经足够。输出如下：
+
+{{ EmbedLiveSample('阻止默认行为', '100%', 180, "", "") }}
```
Comment (yin1999) [inline files/zh-cn/learn/javascript/building_blocks/events/index.md:383]: Just remove the empty params

```suggestion
{{ EmbedLiveSample('阻止默认行为', '100%', 180) }}
```

File: files/zh-cn/learn/javascript/building_blocks/events/index.md [hunk 6]
```diff
@@ -365,157 +362,389 @@ div {
...
+
+{{ EmbedLiveSample('视频播放器示例', '100%', 500) }}
+
+你应该看到，当你点击按钮时，盒子和它所包含的视频都显示出来。但当你点击视频时，视频开始播放，但盒子又被隐藏起来了！
+
+视频在 `<div>` 内（是它的一部分），所以点击视频会*同时*运行两个事件处理器，导致这种行为。
+
+### 使用 stopPropagation() 修复问题
+
+正如我们在上一节所看到的，事件冒泡有时会产生问题，但有一种方法可以防止这些问题。[`Event`](/zh-CN/docs/Web/API/Event)对象有一个可用的函数，叫做 [`stopPropagation()`](/zh-CN/docs/Web/API/Event/stopPropagation)，当在一个事件处理器中调用时，可以防止事件向任何其他元素传递。
```
Comment (jasonren0403) [inline files/zh-cn/learn/javascript/building_blocks/events/index.md:553]:
```suggestion
正如我们在上一节所看到的，事件冒泡有时会产生问题，但有一种方法可以防止这些问题。[`Event`](/zh-CN/docs/Web/API/Event) 对象有一个可用的函数，叫做 [`stopPropagation()`](/zh-CN/docs/Web/API/Event/stopPropagation)，当在一个事件处理器中调用时，可以防止事件向任何其他元素传递。
```

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**